File: programmers/js_src/12940.js
// 최대공약수와 최소공배수

function solution(n, m) {
  var answer = [];
  
  function gcd(n, m) {
      var r = n % m;

      if (r === 0) {
        return m;
      } else {
        return gcd(m, r);
      }      
  }
  
  answer.push(gcd(n, m), n * m / gcd(n, m));
  
  return answer;
}

File: programmers/python/level01/12940.py
# 최대공약수와 최소공배수
def solution(n, m):
    answer = []

    def gcd(n, m):
        r = n % m
        if r == 0:
            return m
        else:
            return gcd(m, r)

    answer.append(gcd(n, m))
    answer.append(n * m / gcd(n, m))

    return answer